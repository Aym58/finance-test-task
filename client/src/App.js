import Container from './components/tickerContainer';

function App() {
	return (
		<>
			<Container />
		</>
	);
}

export default App;
