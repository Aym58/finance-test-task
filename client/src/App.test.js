import React from 'react';
import { render, unmountComponentAtNode } from 'react-dom';
import App from './App';
import { shallow } from 'enzyme';
import { act as domAct } from 'react-dom/test-utils';
import TickerItem from './components/tickerItem';

let container = null;
const data = {
	ticker: 'AAPL',
	exchange: 'NASDAQ',
	price: 279.29,
	change: 64.52,
	change_percent: 0.84,
	dividend: 0.56,
	yield: 1.34,
	last_trade_time: '2021-04-30T11:53:21.000Z',
};

it('renders without crashing', () => {
	shallow(<App />);
});

beforeEach(() => {
	container = document.createElement('div');
	document.body.appendChild(container);
});

afterEach(() => {
	unmountComponentAtNode(container);
	container.remove();
	container = null;
});

it('check TickerItem rendering and date formatting', () => {
	domAct(() => {
		render(<TickerItem data={data} />, container);
	});
	expect(container.querySelector('.ticker-item-time').textContent).toBe(
		'11:53:21'
	);
});
