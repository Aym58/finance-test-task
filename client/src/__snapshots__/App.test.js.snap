// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `
<div
  className="loader-container"
>
  <span
    className="loader"
  />
</div>
`;

exports[`should connect to socket 1`] = `ReactWrapper {}`;

exports[`should select null after timing out 1`] = `ReactWrapper {}`;
